### PR TITLE
Structure change and refactor the code

### PR DESCRIPTION
**Pull request by arshad0010**

### CODE REVIEW

File: src/components/Input/index.tsx [hunk 1]
```diff
@@ -36,107 +35,172 @@ type InputProps = ComponentProps<"input"> & {
   eyeOpen?: React.ReactNode;
   eye?: React.ReactNode;
   prefix?: string;
-  type?: "text" | "password" | "email" | "phone" | "numbers" | "search icon" | "date" | "search" | "Phone" | "number" | "input" | "prefix";
+  type?:
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:38]: use enum

File: src/components/Input/index.tsx [hunk 2]
```diff
@@ -36,107 +35,172 @@ type InputProps = ComponentProps<"input"> & {
...
+    | "Phone"
+    | "number"
+    | "input"
+    | "prefix";
 } & VariantProps<typeof inputStyles>;
 
 export const Input = forwardRef<HTMLInputElement, InputProps>(
-  ({ className, type = "text", search, eyeOpen, eye, prefix, ...props }, ref) => {
+  (
+    { className, type = "text", search, eyeOpen, eye, prefix, ...props },
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:55]: if you already assigned type
then why you used string direct here

File: src/components/Input/index.tsx [hunk 3]
```diff
@@ -36,107 +35,172 @@ type InputProps = ComponentProps<"input"> & {
...
       setValue(newValue);
-      let validationError: string | null = null;
+    };
 
+    const handleBlur = (e: any) => {
+      const newValue = e.target.value;
+      setValue(newValue);
+      let validationError: string | null = null;
+    
       switch (type) {
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:72]: create a function instead of switch

File: src/components/Input/index.tsx [hunk 4]
```diff
@@ -36,107 +35,172 @@ type InputProps = ComponentProps<"input"> & {
...
-          if (newValue && !validateNumbers(newValue)) {
+          if (!validateNumbers(newValue)) {
             validationError = "Please enter a valid number";
           }
           break;
         default:
           break;
       }
-
+
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:91]: remove blank line

File: src/components/Input/index.tsx [hunk 5]
```diff
@@ -36,107 +35,172 @@ type InputProps = ComponentProps<"input"> & {
...
           }
           break;
         default:
           break;
       }
-
+    
       setError(validationError);
     };
+
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:94]: remove blank line

File: src/components/Input/index.tsx [hunk 6]
```diff
@@ -36,107 +35,172 @@ type InputProps = ComponentProps<"input"> & {
...
+        }
+      }
+      return true;
     };
 
     const togglePasswordVisibility = () => {
       setShowPassword((prevShowPassword) => !prevShowPassword);
     };
 
+    function getConditionalStyles(
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:132]: do this mess in another file

File: src/components/Input/index.tsx (hunk 3 above)
Comment (arshad0010) [reply src/components/Input/index.tsx:72]: sir , its already in a function .

File: src/components/Input/index.tsx (hunk 2 above)
Comment (arshad0010) [reply src/components/Input/index.tsx:55]: without this we cannot able to control the input.

File: src/components/Input/index.tsx (hunk 1 above)
Comment (arshad0010) [reply src/components/Input/index.tsx:38]: ok

File: src/components/Input/index.tsx (hunk 6 above)
Comment (arshad0010) [reply src/components/Input/index.tsx]: done

File: src/components/Input/index.stories.tsx [hunk 7]
```diff
@@ -18,89 +19,88 @@ type Story = StoryObj<typeof meta>;
...
+    type: InputType.Text,
     placeholder: "Insert text here",
   },
 };
 
 export const Password: Story = {
   args: {
-    type: "password",
+    type: InputType.Password,
     placeholder: "Password",
```
Comment (deeporiontechnology) [inline src/components/Input/index.stories.tsx:30]: add placeholder in the enum

File: src/components/Input/index.tsx [hunk 8]
```diff
@@ -1,146 +1,142 @@
...
+import {InputType} from "../Input/type";
+import {getConditionalStyles, inputStyles} from "../Input/getStyles"
 
 type InputProps = ComponentProps<"input"> & {
   search?: React.ReactNode;
   eyeOpen?: React.ReactNode;
   eye?: React.ReactNode;
   prefix?: string;
-  type?: "text" | "password" | "email" | "phone" | "numbers" | "search icon" | "date" | "search" | "Phone" | "number" | "input" | "prefix";
+  type?: InputType;
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:12]: type is compulsory

File: src/components/Input/index.tsx [hunk 9]
```diff
@@ -1,146 +1,142 @@
...
       <div className="relative">
-        {type === "prefix" && prefix && (
-          <div className="absolute left-2 top-1/2 transform -translate-y-1/2 pointer-events-none">
-            {prefix}
-          </div>
-        )}
         <input
           ref={ref}
-          type={showPassword && type === "password" ? "text" : type}
+          type={showPassword && type === InputType.Password ? "text" : type}
```
Comment (deeporiontechnology) [inline src/components/Input/index.tsx:126]: this calculation also using for icon
So can we use them in same function

File: src/components/Input/index.stories.tsx (hunk 7 above)
Comment (arshad0010) [reply src/components/Input/index.stories.tsx:30]: ok

**Pull request merged**